File: MetaBrain/imaging/dicom.contract.ts
import type { ImageInput } from "./types";

export type DicomSeries = {
  trace_id: string;
  series_id?: string;
  instance_count?: number;
  safe_metadata: Record<string, unknown>;
};

export type DicomContract = {
  detect_dicom(input: ImageInput): boolean;
  extract_safe_metadata(input: ImageInput): Record<string, unknown>;
  normalize_series(input: ImageInput): Promise<DicomSeries>;
  reject_unsafe(input: ImageInput): { rejected: boolean; reason?: string };
};

export function detectDicomDefensively(input: ImageInput): boolean {
  return input.mime_type.toLowerCase() === "application/dicom" || input.filename?.toLowerCase().endsWith(".dcm") === true;
}

export const DICOM_CONTRACT_STATUS = {
  realDicomParsingImplemented: false,
  dependencyAdded: false,
  enabledByDefault: false,
  note: "No pydicom/MONAI/torch dependency is introduced. Detection is defensive only.",
} as const;

File: MB-Secretaria/imaging_py/dicom_contract.py
from __future__ import annotations

from abc import ABC, abstractmethod
from typing import Any

from .types import ImageInput


class DicomContract(ABC):
    @abstractmethod
    def detect_dicom(self, input_data: ImageInput) -> bool:
        raise NotImplementedError

    @abstractmethod
    def extract_safe_metadata(self, input_data: ImageInput) -> dict[str, Any]:
        raise NotImplementedError

    @abstractmethod
    def normalize_series(self, input_data: ImageInput) -> dict[str, Any]:
        raise NotImplementedError

    @abstractmethod
    def reject_unsafe(self, input_data: ImageInput) -> dict[str, Any]:
        raise NotImplementedError


def detect_dicom_defensively(input_data: ImageInput) -> bool:
    return input_data.mime_type.lower() == "application/dicom" or bool(input_data.filename and input_data.filename.lower().endswith(".dcm"))


DICOM_CONTRACT_STATUS = {
    "real_dicom_parsing_implemented": False,
    "dependency_added": False,
    "enabled_by_default": False,
}
